src/tools/isabelle_executor.py

```diff
@@ -475,10 +527,13 @@ def _parse_proof_context(self, proof_context_str: str, local_hypotheses: typing.
         if len(all_matches) == 0:
             return None
         
-        _, _, _, this_hyps_str, _, goals_str = all_matches[0]
+        context_type, _, _, this_hyps_str, _, goals_str = all_matches[0]
         this_hyps_str, goals_str = this_hyps_str.strip(), goals_str.strip()
         if(goals_str == "No subgoals!"):
             return ProofContext.empty()
+
+        if not found_lemma and not context_type == 'state':
+            raise Exception(f'Error: please provide a full tactic. This step ends in "{context_type}" mode but it should end in "state" mode')
```

Comment:
Isabelle proofs can be in three different modes: `state`, `prove`, and `chain`. Usually (I think), we want to consider only ******full tactics****** — those that begin and end in the `state` mode.

A common issue I saw was ChatGPT generating partial tactics that end in the `prove` or `chain` mode. These tactics still compile, but confuses ChatGPT in the next step. To help prevent this, I require all tactics to be complete and end in the `state` mode (while in proof search mode).